odoo/addons/base/models/res_country.py

```diff
@@ -128,3 +128,21 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
         search_domain.append(('id', 'not in', first_state_ids))
         state_ids = first_state_ids + self._search(search_domain + args, limit=limit, access_rights_uid=name_get_uid)
         return [(state.id, state.display_name) for state in self.browse(state_ids)]
+
+    @api.multi
+    @api.depends('name')
+    def name_get(self):
```

Comment:
it's unclear what you want to achieve here: adding the country code is not really relevant (I suppose?), but if that's really wanted I don't see why you wouldn't want it for len(ids) == 1. 

Also, ideally we always need to keep the name_search() and name_get() in line: it's confusing otherwise

Reply:
@adt asked in the chatter of the task
> Since users can create their own res.country.states, I would suggest overriding the name_get to add the country code between parentheses if the result of the search is > 1.

I don't understand the "in line" thing though

Reply:
too much code for an almost non-existent use case, I suppose: people just have to append themselves the country name in the state name if that occures for them.

about the in-line thing: if you modify the name_get(), you change the display of a record through a m2o. So, to be consistent, you should allow people to search also, in that m2o, on your modification. For example, if we add the country name in the name_get() of states, we need to allow people to search a state by typing the country name in m2o....